main/views.py

```diff
@@ -41,12 +40,16 @@ def get_widget_data(user, widgetlist=[], departments=None):
             ) / context["ipaddress_all"]) * 100)
             context["ipaddress_percentcolor"] = get_progresscolor(context["ipaddress_percent"])
     if "edithistory" in widgetlist:
-        # using exclude(...=other_deps) is much faster than filter(...=departments).distinct()
-        other_deps = Department.objects.all().difference(departments)
-        context['revisions'] = Revision.objects.select_related('user') \
-                                       .prefetch_related('version_set', 'version_set__content_type') \
-                                       .exclude(user__departments__in=other_deps) \
-                                       .order_by("-date_created")[:20]
+        if only_user:
+            revisions = Revision.objects.filter(user=user)
+        else:
+            revisions = Revision.objects.all()
+        context['revisions'] = (
+            revisions
+            .select_related('user')
+            .prefetch_related('version_set', 'version_set__content_type')
+            .order_by("-date_created")[:20]
+        )
     if "newestdevices" in widgetlist:
         if departments:
```

Comment:
that's obsolete now, right?

Reply:
I don't think so. It could be replaced by `if only_user`, but I think this is also fine.